### DISCUSSION TITLE
Can I use chart.js?

### DISCUSSION THREAD

**Discussion by aptnow** [datenstrom/community#585] [Ask a question] [answered]
I would like to put chart data on markdown....
Can I use chart.js with yellow?

**Comment by schulle4u** [comment]
This should be no bigger problem. I suggest to make an extension with the necessary HTML and JS code. Yellow allows the `<canvas>` element in pages, however you will have to put the inline JS part in an external file. You can try out the [Helloworld extension](https://github.com/schulle4u/yellow-extension-helloworld) as a starting point, it demonstrates how to create a simple shortcut and how Yellow handles inclusion of JS code.

**Reply by schulle4u** [reply]
OK I had a closer look at the usage example. It seems that you'll have to find a way to parse the chart data in a more practical way. The example suggests to make it a part of the JS code, but this is not very handy for an extension. A JSON file could work, maybe other developers are more familiar to this.

**Reply by markseuffert** [reply]
I think an extension could solve this in two ways:

1. Reference an external file that contains the JSON chart data.
2. Use a Markdown code block for the JSON chart data.

Once you have the JSON chart data in your extension, you can make it available to JavaScript. Many extensions generate a div element with data-attributes. In this case there's a bigger amount of data, so you probably want to create a canvas element and insert the (sanitised) JSON chart data into the page header. Some glue code in JavaScript to bring everything together.

**Comment by GiovanniSalmeri** [comment, marked as answer]
@aptnow Just a quick and dirty extension that allows to use Chartjs: https://github.com/GiovanniSalmeri/yellow-chartjs

I have followed the suggestions by @markseu together with a DOM manipulation which leaves the code very simple. It would be better if the data could be inserted in a format more human-writable than JSON, but Chartjs should already be perfectly usable in Yellow now.

**Reply by GiovanniSalmeri** [reply]
Now the extension, also thanks to the advice of @markseu, is cleaner and safer. The same approach should work for several commonly used JS libraries (and there are fewer and fewer excuses not to use Yellow 😊).
